src/frontend/data/queries/updateOne.spec.tsx

```diff
@@ -0,0 +1,144 @@
+import fetchMock from 'fetch-mock';
+import React from 'react';
+
+import { updateOne } from './updateOne';
+
+let mockAppData = {};
+jest.mock('../appData', () => ({
+  get appData() {
+    return mockAppData;
+  },
+}));
+
+describe('queries/updateOne', () => {
+  beforeEach(() => (mockAppData = {}));
+  afterEach(() => fetchMock.restore());
+
+  it('updates the resource, handles the response and resolves with a success', async () => {
+    mockAppData = { jwt: 'some token' };
+    const objectToUpdate = { objets: 'data' };
+    fetchMock.mock('/api/model-name/1/', { key: 'value' });
+
+    const response = await updateOne({
+      name: 'model-name',
+      id: '1',
+      object: objectToUpdate,
+    });
+
+    expect(fetchMock.lastCall()![0]).toEqual('/api/model-name/1/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        Authorization: 'Bearer some token',
+        'Content-Type': 'application/json',
+      },
+      method: 'PUT',
+      body: JSON.stringify(objectToUpdate),
+    });
+    expect(response).toEqual({ key: 'value' });
+  });
+
+  it('updates the resource without JWT token', async () => {
+    mockAppData = {};
+    const objectToUpdate = { objets: 'data' };
+    fetchMock.mock('/api/model-name/1/', { key: 'value' });
+
+    const response = await updateOne({
+      name: 'model-name',
+      id: '1',
+      object: objectToUpdate,
+    });
+
+    expect(fetchMock.lastCall()![0]).toEqual('/api/model-name/1/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'PUT',
+      body: JSON.stringify(objectToUpdate),
+    });
+    expect(response).toEqual({ key: 'value' });
+  });
+
+  it('resolves with a failure and handles it when it fails to update the resource (local)', async () => {
+    mockAppData = { jwt: 'some token' };
+    const objectToUpdate = { objets: 'data' };
+    fetchMock.mock(
+      '/api/model-name/1/',
+      Promise.reject(new Error('Failed to perform the request')),
+    );
+
+    await expect(
+      updateOne({
+        name: 'model-name',
+        id: '1',
+        object: objectToUpdate,
+      }),
+    ).rejects.toThrowError('Failed to perform the request');
+
+    expect(fetchMock.lastCall()![0]).toEqual('/api/model-name/1/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        Authorization: 'Bearer some token',
+        'Content-Type': 'application/json',
+      },
+      method: 'PUT',
+      body: JSON.stringify(objectToUpdate),
+    });
+  });
+
+  it('resolves with a 404 and handles it when it fails to update the resource (api)', async () => {
+    mockAppData = { jwt: 'some token' };
+    const objectToUpdate = { objets: 'data' };
+    fetchMock.mock('/api/model-name/1/', 404);
+
+    let thrownError;
+
+    try {
+      await updateOne({
+        name: 'model-name',
+        id: '1',
+        object: objectToUpdate,
+      });
+    } catch (error) {
```

Comment:
not possible to do this ?
```
await expect(
updateOne({
        name: 'model-name',
        id: '1',
        object: objectToUpdate,
      })
).rejects.toThrowError({ code: 'exception' });

```

Reply:
Seems like error thrown here https://github.com/openfun/marsha/pull/1019/files#diff-897a27888c0af44565d0da0b82d2df7b260c7cf17a0526e9b6f89a3b89c9938fR23 (shamelessly taken from createOne) is not catch by `.rejects.toThrowError`

```  ● queries/updateOne › resolves with a 404 and handles it when it fails to update the resource (api)

    expect(received).rejects.toThrowError(expected)

    Expected message: undefined

    Received function did not throw

      106 |       id: '1',
      107 |       object: objectToUpdate,
    > 108 |     })).rejects.toThrowError({ code: 'exception' });
          |                 ^
      109 |
      110 |     // expect(thrownError).toEqual({ code: 'exception' });
      111 |
```

Reply:
ok it seems you can't use `toThrow` with an expected object.